File: Baekjoon/Class3/04.11047 동전 0.java
import java.io.*;
import java.util.StringTokenizer;

class Main {
    public static void main(String[] args) throws IOException {
        int n, k, coinCount = 0;
        int[] coins;
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        n = Integer.parseInt(st.nextToken());
        k = Integer.parseInt(st.nextToken());

        coins = new int[n]; // 동전 종류
        for(int i = n - 1; i >= 0; i--)
            coins[i] = Integer.parseInt(br.readLine());

        for(int coin: coins) { // 동전 개수
            if(k == 0) break;
            coinCount += k / coin;
            k %= coin;
        }
        System.out.print(coinCount);
        br.close();
    }
}

File: Baekjoon/Class3/04.11047 동전 0.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int n, k;
    cin >> n >> k;

    vector<int> coins(n); // 동전 종류
    for (int i = n - 1; i >= 0; i--)
        cin >> coins[i];

    int coin_count = 0; // 동전 개수
    for (const int coin : coins) {
        if (k == 0) break;
        coin_count += k / coin;
        k %= coin;
    }
    cout << coin_count;
    return 0;
}